osu.Game.Rulesets.Osu/Mods/OsuModClassic.cs

```diff
@@ -59,7 +59,7 @@ public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset
             if (ClassicNoteLock.Value)
                 osuRuleset.Playfield.HitPolicy = new ObjectOrderedHitPolicy();
 
-            hiddenModActive = drawableRuleset.Mods.OfType<ModHidden>().Any();
+            usingHiddenFading = !drawableRuleset.Mods.OfType<OsuModHidden>().FirstOrDefault()?.OnlyFadeApproachCircles.Value ?? false;
```

Comment:
I don't quite understand this change. What's the significance of approach circles in this logic?

Reply:
If `OnlyFadeApproachCircles` is true, then the main circle piece _isn't_ being affected by hidden and should fade out according to stable logic, shouldn't it?

Reply:
Right, I see. lots of inverse conditions are confusing.

Reply:
This should be written as `... != true` instead to reduce the confusion factor. Also should probably be using `SingleOrDefault()` to retrieve the mod instance.

Reply:
Ok so I misread and as a result https://github.com/ppy/osu/pull/22543/commits/58d64cdbd04e07f56c50913efab939b664204751 was arguably even worse than before. After drawing up truth tables for this I *believe* that what was wanted here is actually `... == false` which I've applied with test coverage (mostly visual, animation timings are too bothersome to exercise in tests). Please double-check though.

Reply:
The logic works. I'm a uncertain about `TestHitCircleClassicAndFullHiddenMods` since `alphaAtMiss` will be zero regardless of the classic mod, but if you think its fine then whatever.

Reply:
I know, as I said, I added that mostly for visual reference. If I was confident in a way to easily and reliably test that the circle fades out prior to miss judgement then I would.